smartsim/_core/mli/message_handler.py

```diff
@@ -401,8 +401,18 @@ def serialize_request(request: request_capnp.RequestBuilder) -> bytes:
 
         :param request: Request to be serialized
         :returns: Serialized request bytes
+        :raises ValueError: If serialization fails
         """
-        return request.to_bytes()
+        display_name = request.schema.node.displayName  # type: ignore
+        class_name = display_name.split(":")[-1]
+        if class_name != "Request":
+            raise ValueError(
+                "Error serializing the request. Value passed in is not a request"
```

Comment:
we should log identifying info, where possible, to help fix the defect. we might want the `class_name` value here so we know where the mistake came from.

ditto on others, like .... in load_model, `model_id.key`, `device`, etc

Reply:
To be super clear, you mean like adding like `f"Error serializing the request. Value passed in is not a request: {request}"` in this example, and not using the actual logger to log more stuff?